elastic/kibana
# Fix panel content styling

The panel content styling was causing a bit weird behavior when toggling the legend.

Whatever element was rendered as content of a panel, got via the `.panel-content` a `flex: 1 1 100%`. That will cause the content to become 100% anyway, but it will allow to grow (also over the border of the panel). This is causing the visualize legend when opened (before the chart actually resizes), to exceed the panels dimension and overlap the panel beside it:

![screenshot-20171222-174829](https://user-images.githubusercontent.com/877229/34305972-1313b102-e741-11e7-9cd6-fc994cb37e05.png)

I introduced a new class `.panel-content--fullWidth` that can be applied by the embeddable if it wants its element to always have `width: 100%` - in which case the issue doesn't appear anymore. I thought about adding it to the `.panel-content` class directly, but this wouldn't work if an embeddable would render more then one DOM elements as its root. We currently don't have that case, but that way we are safe if an embeddable would ever render multiple DOM nodes on its root level (it just doesn't use this class).

**How to test**

* Place a chart with a legend on the dashboard
* Close that legend and open it
* See how the legend exceeds the panel for a short amount of time while it's opening

With this fix, this shouldn't occur anymore.
